oceanwaves/spectral.py

```diff
@@ -97,19 +128,49 @@ def directional_spreading(theta, theta_peak=0., s=20., units='deg',
         raise ValueError('Unknown units: %s')
 
     # compute directional spreading
-    #A1 = (2.**s) * (gamma(s / 2 + 1))**2. / (np.pi * gamma(s + 1))
-    #p_theta = A1 * np.maximum(0., np.cos(theta - theta_peak))
+    # A1 = (2.**s) * (gamma(s / 2 + 1))**2. / (np.pi * gamma(s + 1))
+    # p_theta = A1 * np.maximum(0., np.cos(theta - theta_peak))
     p_theta = np.maximum(0., np.cos(theta - theta_peak))**s
 
     # convert to original units
     if units.lower().startswith('deg'):
         theta = np.degrees(theta)
         theta_peak = np.degrees(theta_peak)
         p_theta = np.degrees(p_theta)
-    
+
     # normalize directional spreading
     if normalize:
         p_theta /= trapz_and_repeat(p_theta, theta - theta_peak, axis=-1)
 
     return p_theta
 
+
+def _check_dtype(var):
+    '''
+    Auxiliary function to detect and fix dtypes, if needed
+
+    Parameters
+    ----------
+    var : anything
+        Array to check
+
+    Returns
+    -------
+    var : numpy.ndarray
+       The same as the input but either as a float or a array of floats
+    '''
+    # fist, if it's a list, convert to a numpy.ndarray
+    if type(var) == list:
+        var = np.array(var)
+    # if it's an np.ndarray(), make sure it's a array of floats
+    elif type(var) == np.ndarray:
+        if var.dtype != np.dtype('float64'):
```

Comment:
Use `np.float` instead of `np.float64`. That way the user can set the required precision. Also, `np.float64` is a non-native type, which we generally don't want to use as default.